node-example/runtime/src/lib.rs

```diff
@@ -274,6 +274,36 @@ impl pallet_template::Trait for Runtime {
 	type Event = Event;
 }
 
+parameter_types! {
+	pub const HashWeight: Weight = 1;
+}
+
+type MmrHash = <Keccak256 as sp_runtime::traits::Hash>::Output;
+
+/// A BEEFY consensus digest item with MMR root hash.
+pub struct DepositLog;
+impl pallet_mmr::primitives::OnNewRoot<MmrHash> for DepositLog {
+	fn on_new_root(root: &Hash) -> Weight {
+		let digest = DigestItem::Consensus(
+			beefy_primitives::BEEFY_ENGINE_ID,
+			codec::Encode::encode(&beefy_primitives::ConsensusLog::<BeefyId>::MmrRoot(*root)),
+		);
+		<frame_system::Module<Runtime>>::deposit_log(digest);
+
+		<Runtime as frame_system::Trait>::DbWeight::get().writes(1)
```

Comment:
This was a bit surprising to me but indeed `fn deposit_log` is documented as being `O(1)` and performing 1 storage write. Not sure if this is accurate though since the storage write is only ephemeral (the entry is killed on block finalization) and the changes aren't actually committed to disk. Anyway I think this is fine as is :)

Reply:
Yeah, this thing is flawed anyway, I'm currently rewriting substrate PR to actually have `on_initialize` weight benchmarked instead of hardcoded, so will update here soon as well.